src/api/helpers/formatEntityData.ts

```diff
@@ -99,6 +99,82 @@ export function getWorkBasicInfo(work: Record<string, unknown> | null | undefine
 		};
 }
 
+/**
+ * A function to extract structured author credit details from an Edition ORM entity.
+ * @function
+ * @param {object} edition - an edition ORM entity
+ * @returns {object|null} an object containing the author credit data, or null if not available.
+ *
+ * @example
+ * 		const edition = await orm.func.entity.getEntity(orm, 'Edition', bbid, relations);
+ *		filterAuthorCredit(edition);
+ *		/* => {
+ *			"authorCount": 1,
+ *			"id": 135,
+ *			"names": [
+ *				{
+ *					"authorBBID": "e5c4e68b-bfce-4c77-9ca2-0f0a2d4d09f0",
+ *					"authorCreditID": 135,
+ *					"name": "J. K. Rowling"
+ *				}
+ *			]
+ *		}
+ */
+
+export function filterAuthorCredit(edition: any) {
+	if (_.isNil(edition) || _.isNil(edition.authorCredit)) { return null; }
+
+	const credit = edition.authorCredit;
+	return {
+		authorCount: credit.authorCount || null,
+		id: credit.id || null,
+		names: (credit.names || []).map(({authorCreditID, authorBBID, name}) => ({
```

Comment:
Not sure what is happening here, but it looks like the mapping function just returns the same object structure, so is there a point in it?

Reply:
The objects in names array have 5 properties:
1. authorCreditId
2. positon
3. authorBBID
4. name
5. joinphrase
and I felt we dont need the postion and joinphrase. So I am removing them in the mapping function.

Reply:
I see, thanks for the clarifications.
I think we will need the position and joinphrase to re-compose the full author credit phrase.

For example (choosing a random recent entry with more than one author): https://bookbrainz.org/edition/c82ed17a-4627-48d1-9265-1b0a8ef34b37
Here if you had your own website where you wanted to present this edition with the author credits, you would be left with an array of names ["Daniel Keel","Daniel Kampa"], hopefully in the original order (not guaranteed).
With the position and joinphrase you can reconstitute "Daniel Keel und Daniel Kampa" like we do on the BB website itself.